code/game/objects/structures/window.dm

```diff
@@ -514,12 +514,35 @@
 	var/id = 0
 	var/active = 0
 
+/obj/machinery/button/windowtint/New(turf/loc, w_dir=null)
```

Comment:
If you haven't already tried, can you switch this to use `Initialize` and see if it still works properly? Just don't want to add anymore unnecessary New's.